### PR TITLE
Style and remove "Save" button for Default Room Settings

### PR DESCRIPTION
**Pull request by nickmurray47**
### Details
"Save" button is hidden for default/archived rooms and updated the style.

### Fixed Issues

$ [GH_LINK](https://github.com/Expensify/App/issues/8146)

### Tests / QA Steps
1. See default room - either #announce or #admins
1. Click room avatar
1. Click settings
2. Verify "Save" button is no longer visible and text inputs are just text.
3. Click "+" Global create, create a new room, add name and in the new room's settings, verify you can edit and save the name of the room but not the workspace.

#### Contributor (PR Author) Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I clearly indicated the environment tests should be run in (Staging vs Production)
- [x] I wrote testing steps that cover success & fail scenarios (if applicable)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I added comments when the code was not self explanatory
    - [x] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [x] I followed proper naming convention for platform-specific files (if applicable)
    - [x] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
    - [x] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I corroborated the UI performance was not affected (the performance is the same than `main` branch)
- [x] If I created a new component I verified that a similar component doesn't exist in the codebase

#### PR Reviewer Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [x] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase

#### Web
default room
Screen Shot 2022-03-15 at 1 35 23 PM

new room
Screen Shot 2022-03-16 at 3 25 02 PM

#### Mobile Web
Screen Shot 2022-03-16 at 12 39 55 PM
Screen Shot 2022-03-16 at 12 55 20 PM

#### Desktop
Screen Shot 2022-03-15 at 2 45 38 PM
Screen Shot 2022-03-16 at 3 25 02 PM

#### iOS
Screen Shot 2022-03-16 at 12 20 25 PM
Screen Shot 2022-03-16 at 3 10 41 PM

#### Android
Screen Shot 2022-03-15 at 3 05 33 PM
Screen Shot 2022-03-16 at 4 11 15 PM

### CODE REVIEW

**Comment by nickmurray47** [comment]
cc @shawnborton what do you think of the initial Web screenshot?

**Comment by shawnborton** [comment]
Just to make sure I am following, in this case the room name input is not editable right? Same with the workspace input?

**Comment by nickmurray47** [comment]
>  in this case the room name input is not editable right? Same with the workspace input?

Correct, they are both not editable. Should we make that clear to the end-user? We could gray the text box out.

**Comment by shawnborton** [comment]
I was thinking we'd use this pattern when the input is read-only and can't be edited:
image

**Comment by nickmurray47** [comment]
got it, will update

**Comment by nickmurray47** [comment]
Screen Shot 2022-03-15 at 1 10 44 PM

**Comment by shawnborton** [comment]
So close! Final details would be that the total height of the small label + value should be 40px:
image

And we get that by making that upper label have a line height of 16px, a bottom margin of 4px, and then the value below has a line height of 20px.

**Comment by nickmurray47** [comment]
How do we feel about the lefthand spacing? Also looks a little off to me but not sure if that's intended.

File: src/pages/ReportSettingsPage.js [hunk 1]
```diff
@@ -151,7 +151,7 @@ class ReportSettingsPage extends Component {
 
     render() {
         const shouldShowRename = !ReportUtils.isPolicyExpenseChat(this.props.report);
-        const shouldDisableRename = ReportUtils.isDefaultRoom(this.props.report)
+        const isDefaultOrArchivedRoom = ReportUtils.isDefaultRoom(this.props.report)
```
Comment (bondydaa) [inline src/pages/ReportSettingsPage.js:154]: NAB but I personally think the previous variable name was more descriptive and explained why we cared these checks.

File: src/pages/ReportSettingsPage.js [hunk 2]
```diff
@@ -186,39 +181,50 @@ class ReportSettingsPage extends Component {
...
+                            <View style={[styles.flexRow]}>
                                 <View style={[styles.flex3]}>
-                                    <RoomNameInput
-                                        initialValue={this.state.newRoomName}
-                                        policyID={linkedWorkspace && linkedWorkspace.id}
-                                        errorText={this.state.errors.newRoomName}
-                                        onChangeText={newRoomName => this.clearErrorAndSetValue('newRoomName', newRoomName)}
-                                        disabled={shouldDisableRename}
-                                    />
+                                    {!isDefaultOrArchivedRoom ? (
```
Comment (bondydaa) [inline src/pages/ReportSettingsPage.js:186]: also NAB but the !boolean always throws my brain for a loop and makes me have to think hard about which thing is getting rendered so personally I think it's easier to reason about the code if it was
```
boolean ? renderIfTrue : renderIfFalse;
```

but not the end of the world if you want to leave it the way you have it.

**Review by bondydaa — DISMISSED** [review]
no blockers from me, going to tap someone else who's actually written App code though too.

File: src/pages/ReportSettingsPage.js (hunk 1 above)
Comment (nickmurray47) [reply src/pages/ReportSettingsPage.js:154]: that's fair, I think it was confusing for me when actually used, i.e. a boolean where `!shouldDisableRename` was like a weird double negative in my head

File: src/pages/ReportSettingsPage.js (hunk 2 above)
Comment (nickmurray47) [reply src/pages/ReportSettingsPage.js:186]: definitely agree here! was just about to make this change

**Comment by nickmurray47** [comment]
screenshots and review comments updated, gonna pull in one more App reviewer!

**Comment by shawnborton** [comment]
How exactly does the visibility: restricted stuff work? I feel like that should look identical to the sections above it where "Visibility" is in the smaller gray text.

**Comment by nickmurray47** [comment]
How does this look @shawnborton?
Screen Shot 2022-03-16 at 12 55 20 PM

**Comment by shawnborton** [comment]
Looks great!

**Comment by nickmurray47** [comment]
checks pass, ready for review @stitesExpensify @bondydaa

**Comment by shawnborton** [comment]
Nice, I think you just need to update the screenshots given the "Visibility" changes and then we're good.

**Review by bondydaa — APPROVED** [review]

**Review by stitesExpensify — APPROVED** [review]

**Comment by nickmurray47** [comment]
screenshots updated, gonna merge soon

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
I'll go ahead and merge this one.

**Pull request merged**